src/sas/sascalc/corfunc/vectorisation.py

```diff
@@ -0,0 +1,20 @@
+import numpy as np
+from typing import Union, Callable, Any
+
+# TODO: Proper typing when newer numpy (>1.20) is supported
+ArrayLike = Union[np.ndarray, float, int]
+
+def assure_vectorised_argument(f: Callable):
```

Comment:
I imagine that there will be many places in the code where this could be useful.  Maybe we could move it to a more general location than just within Corfunc?

Reply:
Rough plan is to generalise it (or find the code where I've done it before) and to move it somewhere better, but the util section needs tidying in general. I've not round to doing this yet, will create a separate issue for it.

Reply:
Created issue #2054